### PR TITLE
Add RTPTransceiver.Mid()

### LINKED ISSUE #1148: Order of Transceiver / Track association sometimes changes after track is removed
Could be related to my changes in https://github.com/pion/webrtc/pull/1140, since this happens after I remove a track.

### Your environment.
  - Version: d5dbe1f
  - Browser: N/A

### What did you do?

I called `peerConnection.RemoveTrack()` and renegotiated.

### What did you expect?

I expected `a=mid`/`a=ssrc` associations to stay the same.

### What happened?
Looks like the order of `a=mid`/`a=ssrc` changed.

### Before SDP

I've removed the `a=candidate` and `a=rtpmap` lines, and added spaces between `m=` sections to make the SDPs easier to read. 

An offer before the track `sfu_{82cc6157-3738-3447-9952-93bcf6adbe32}` (ssrc: `197955246`) was removed:

```
2020-04-18T08:48:26.569854+02:00 [            sdp] [VDYSr00cAfqrnx02zsxjA6] Local signal.type: offer, signal.sdp: v=0
o=- 842954722 1587192506 IN IP4 0.0.0.0
s=-
t=0 0
a=fingerprint:sha-256 86:CF:28:50:97:06:CB:AB:06:4B:33:20:57:85:E3:9E:DF:81:B2:B3:A0:94:E4:C9:67:41:2E:12:D7:32:BF:7A
a=group:BUNDLE 0 1 2 3 4 5

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:0
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:2596996162 cname:UaSxtEzjIuBvsjNK
a=ssrc:2596996162 msid:UaSxtEzjIuBvsjNK ZaYuzSRctwZRCnEN
a=ssrc:2596996162 mslabel:UaSxtEzjIuBvsjNK
a=ssrc:2596996162 label:ZaYuzSRctwZRCnEN
a=msid:UaSxtEzjIuBvsjNK ZaYuzSRctwZRCnEN
a=sendrecv
a=end-of-candidates

m=audio 9 UDP/TLS/RTP/SAVPF 111 0 8 9
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:1
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:111 minptime=10;useinbandfec=1
a=ssrc:4039455774 cname:nTJUWbdJlBYrngSL
a=ssrc:4039455774 msid:nTJUWbdJlBYrngSL nvjHciImKOcWliIm
a=ssrc:4039455774 mslabel:nTJUWbdJlBYrngSL
a=ssrc:4039455774 label:nvjHciImKOcWliIm
a=msid:nTJUWbdJlBYrngSL nvjHciImKOcWliIm
a=sendrecv
a=end-of-candidates

m=application 9 DTLS/SCTP 5000
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:2
a=sendrecv
a=sctpmap:5000 webrtc-datachannel 1024
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=end-of-candidates

m=audio 9 UDP/TLS/RTP/SAVPF 111 0 8 9
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:3
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:111 minptime=10;useinbandfec=1
a=ssrc:1961657070 cname:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1961657070 msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=ssrc:1961657070 mslabel:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1961657070 label:sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=sendrecv
a=end-of-candidates

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:4
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:1232322661 cname:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1232322661 msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=ssrc:1232322661 mslabel:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1232322661 label:sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=sendrecv
a=end-of-candidates

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:5
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:197955246 cname:sfu_lIq38Rktc6gygdHB73mvz1_{e1ad4f34-c939-d240-9a70-99cb5c32c4ab}
a=ssrc:197955246 msid:sfu_lIq38Rktc6gygdHB73mvz1_{e1ad4f34-c939-d240-9a70-99cb5c32c4ab} sfu_{82cc6157-3738-3447-9952-93bcf6adbe32}
a=ssrc:197955246 mslabel:sfu_lIq38Rktc6gygdHB73mvz1_{e1ad4f34-c939-d240-9a70-99cb5c32c4ab}
a=ssrc:197955246 label:sfu_{82cc6157-3738-3447-9952-93bcf6adbe32}
a=msid:sfu_lIq38Rktc6gygdHB73mvz1_{e1ad4f34-c939-d240-9a70-99cb5c32c4ab} sfu_{82cc6157-3738-3447-9952-93bcf6adbe32}
a=sendrecv
a=end-of-candidates
```

### After SDP

An offer after the track `sfu_{82cc6157-3738-3447-9952-93bcf6adbe32}` (ssrc: `197955246`) was removed:

```
2020-04-18T08:48:34.830186+02:00 [            sdp] [VDYSr00cAfqrnx02zsxjA6] Local signal.type: offer, signal.sdp: v=0
o=- 104360727 1587192514 IN IP4 0.0.0.0
s=-
t=0 0
a=fingerprint:sha-256 86:CF:28:50:97:06:CB:AB:06:4B:33:20:57:85:E3:9E:DF:81:B2:B3:A0:94:E4:C9:67:41:2E:12:D7:32:BF:7A
a=group:BUNDLE 0 1 2 3 4 5

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:0
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=recvonly
a=end-of-candidates

m=audio 9 UDP/TLS/RTP/SAVPF 111 0 8 9
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:1
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:111 minptime=10;useinbandfec=1
a=ssrc:4039455774 cname:nTJUWbdJlBYrngSL
a=ssrc:4039455774 msid:nTJUWbdJlBYrngSL nvjHciImKOcWliIm
a=ssrc:4039455774 mslabel:nTJUWbdJlBYrngSL
a=ssrc:4039455774 label:nvjHciImKOcWliIm
a=msid:nTJUWbdJlBYrngSL nvjHciImKOcWliIm
a=sendrecv
a=end-of-candidates

m=application 9 DTLS/SCTP 5000
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:2
a=sendrecv
a=sctpmap:5000 webrtc-datachannel 1024
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=end-of-candidates

m=audio 9 UDP/TLS/RTP/SAVPF 111 0 8 9
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:3
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:111 minptime=10;useinbandfec=1
a=ssrc:1961657070 cname:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1961657070 msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=ssrc:1961657070 mslabel:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1961657070 label:sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{033036ce-58f6-2046-997d-a3cf2dbb731a}
a=sendrecv
a=end-of-candidates

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:4
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:2596996162 cname:UaSxtEzjIuBvsjNK
a=ssrc:2596996162 msid:UaSxtEzjIuBvsjNK ZaYuzSRctwZRCnEN
a=ssrc:2596996162 mslabel:UaSxtEzjIuBvsjNK
a=ssrc:2596996162 label:ZaYuzSRctwZRCnEN
a=msid:UaSxtEzjIuBvsjNK ZaYuzSRctwZRCnEN
a=sendrecv
a=end-of-candidates

m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:5
a=ice-ufrag:zQrcqINzoemkGBQB
a=ice-pwd:vUWlevctbIMZJLaJjVBbccKjVVXwyZDj
a=rtcp-mux
a=rtcp-rsize
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:1232322661 cname:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1232322661 msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=ssrc:1232322661 mslabel:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c}
a=ssrc:1232322661 label:sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=msid:sfu_lIq38Rktc6gygdHB73mvz1_{24fa7a57-1305-5344-90df-4eed95974f5c} sfu_{c8092227-628f-4941-a414-6cee6bc9cb33}
a=sendrecv
a=end-of-candidates
```

### PR DESCRIPTION
**Pull request by jeremija**
#### Description
@Sean-Der - just wanted to share this before I call it a night. I'm not sure if you had other plans regarding issue #1148, so I'd love to hear your thoughts.

#### Reference issue
(Hopefully) Fixes #1148

### CODE REVIEW

**Comment by Sean-Der** [comment]
@jeremija This conceptually LGTM! 

So you are just making sure to store the mid, so when sections go inactive/change we don't move around?

File: rtptransceiver.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 
 // RTPTransceiver represents a combination of an RTPSender and an RTPReceiver that share a common mid.
 type RTPTransceiver struct {
+	mid       atomic.Value
```
Comment (Sean-Der) [inline rtptransceiver.go:12]: ` // string` just so the reader knows the type

File: rtptransceiver.go [hunk 2]
```diff
@@ -39,6 +40,29 @@ func (t *RTPTransceiver) Receiver() *RTPReceiver {
...
+	}
+	t.mid.Store(mid)
+	return nil
+}
+
+// Mid get the Transceiver's mid value. If not already set, this value will be set in CreateOffer or CreateAnswer.
+func (t *RTPTransceiver) Mid() string {
+	value := t.mid.Load()
+	if v := t.mid.Load(); v != nil {
+		return value.(string)
```
Comment (Sean-Der) [inline rtptransceiver.go:56]: return `v.(string)`

**Comment by jeremija** [comment]
> So you are just making sure to store the mid, so when sections go inactive/change we don't move around?

Yes, that's about it. Perhaps it should error out when a transceiver is found by Mid, but with wrong kind and/or codec?

**Comment by sgotti** [comment]
@jeremija just discovered the same issue some days ago and was ready to open a PR when noticed this one. If you're interested on my comment description, implementation and the tests I implemented you can find them here: https://github.com/sgotti/webrtc/commit/3b82f999107c0fb3b9b757ce991a9c250ad05909

Thanks!

**Comment by jeremija** [comment]
Thanks for sharing, @sgotti! Will definitely take a look!

**Comment by Sean-Der** [comment]
@jeremija This looks amazing to me! I think we should get @sgotti added to the org (if you are interested!) and you guys can  own this :)

I have reduced availability till the weekend, and I don't want to block you :(

**Pull request merged**